### PR TITLE
[plugin.video.raitv] 3.1.1

### PR DESCRIPTION
**Pull request by maxbambi**
### Description
New version from Nightflyer's "Rai on demand".
This plugin was deleted from Official Repo, because broken and not updated by its author.
Now it is updated and working with new changes in Raiplay web site.

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.raitv/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*.py[co]
```
Comment (enen92) [inline plugin.video.raitv/.gitignore]: Drop from pr pls

File: plugin.video.raitv/addon.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    <import addon="script.common.plugin.cache" version="2.5.0"/>
+  </requires>
+  <extension point="xbmc.python.pluginsource"
+            library="default.py">
+        <provides>video audio</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <language>it</language>
+    <platform>all</platform>
+    <summary lang="it">I programmi delle reti Rai e i servizi interattivi</summary>
```
Comment (enen92) [inline plugin.video.raitv/addon.xml]: it_IT and en_GB

File: plugin.video.raitv/changelog.txt [hunk 3]
```diff
@@ -0,0 +1,98 @@
+[B]3.1.1[/B]
```
Comment (enen92) [inline plugin.video.raitv/changelog.txt]: Changelog is no longer parsed by kodi. Please use the news tag in add-on xml

File: plugin.video.raitv/default.py [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+
+def show_ondemand_items(url):
+    xbmc.log("ContentSet URL: " + url)
+    raiplay = RaiPlay()
+    items = raiplay.getContentSet(url)
+    for item in items:
+        title = item["name"]
+        if "subtitle" in item and item["subtitle"] != "" and item["subtitle"] != item["name"]:
+            title = title + " (" + item["subtitle"] + ")"
+        liStyle = xbmcgui.ListItem(title, thumbnailImage=raiplay.getThumbnailUrl(item["images"]["landscape"]))
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Drop

File: plugin.video.raitv/default.py [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+        liStyle.setInfo("video", {})
+        addLinkItem({"mode": "play",
+            "path_id": item["pathID"]}, liStyle)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+    
+def search_ondemand_programmes():
+    kb = xbmc.Keyboard()
+    kb.setHeading("Cerca un programma")
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Localize

File: plugin.video.raitv/default.py [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+
+def show_ondemand_programme(pathId):
+    xbmc.log("PathID: " + pathId)
+    raiplay = RaiPlay()
+    programme = raiplay.getProgramme(pathId)
+    
+    if (len(programme["infoProg"]["tipologia"]) > 0) and programme["infoProg"]["tipologia"][0]["nome"] == "Film":
+        if "pathFirstItem" in programme:
+            liStyle = xbmcgui.ListItem(programme["infoProg"]["name"], thumbnailImage=raiplay.getThumbnailUrl(programme["infoProg"]["images"]["landscape"]))
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Drop

File: plugin.video.raitv/default.py [hunk 7]
```diff
@@ -0,0 +1,525 @@
...
+        addDirectoryItem({"mode": "ondemand_list", "index": chr(ord('A')+i), "path_id": pathId}, liStyle)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+
+def show_ondemand_index(index, pathId):
+    xbmc.log("PathID: " + pathId)
+    xbmc.log("Index: " + index)
+    raiplay = RaiPlay()
+    dir = raiplay.getProgrammeList(pathId)
+    for item in dir[index]:
+        liStyle = xbmcgui.ListItem(item["name"], thumbnailImage=raiplay.getThumbnailUrl(item["images"]["landscape"]))
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Same

File: plugin.video.raitv/default.py [hunk 8]
```diff
@@ -0,0 +1,525 @@
...
+def show_ondemand_programmes(pathId):
+    xbmc.log("PathID: " + pathId)
+    raiplay = RaiPlay()
+    blocchi = raiplay.getCategory(pathId)
+
+    if len(blocchi) > 1:
+        xbmc.log("Blocchi: " + str(len(blocchi)))
+        
+    for item in blocchi[0]["lanci"]:
+        liStyle = xbmcgui.ListItem(item["name"], thumbnailImage=raiplay.getThumbnailUrl(item["images"]["landscape"]))
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Drop thumbnail, use setArt

File: plugin.video.raitv/default.py [hunk 9]
```diff
@@ -0,0 +1,525 @@
...
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+    
+def show_ondemand_root():
+    raiplay = RaiPlay()
+    items = raiplay.getMainMenu()
+    for item in items:
+        if item["sub-type"] in ("RaiPlay Tipologia Page", "RaiPlay Genere Page", "RaiPlay Tipologia Editoriale Page" ):
+            liStyle = xbmcgui.ListItem(item["name"])
+            addDirectoryItem({"mode": "ondemand", "path_id": item["PathID"], "sub_type": item["sub-type"]}, liStyle)
+    liStyle = xbmcgui.ListItem("Cerca")
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Same

File: plugin.video.raitv/default.py [hunk 10]
```diff
@@ -0,0 +1,525 @@
...
+            liStyle.setInfo("audio", {})
+            addLinkItem({"mode": "play",
+                "path_id": audioUrl}, liStyle)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+    
+def show_ondemand_root():
+    raiplay = RaiPlay()
+    items = raiplay.getMainMenu()
+    for item in items:
+        if item["sub-type"] in ("RaiPlay Tipologia Page", "RaiPlay Genere Page", "RaiPlay Tipologia Editoriale Page" ):
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Localize these strings pls

File: plugin.video.raitv/default.py [hunk 11]
```diff
@@ -0,0 +1,525 @@
...
+        
+        if audioUrl is None:
+            # programme is not available
+            liStyle = xbmcgui.ListItem(startTime + " [I]" + title + "[/I]",
+                thumbnailImage=thumb)
+            liStyle.setInfo("audio", {})
+            addLinkItem({"mode": "nop"}, liStyle)
+        else:
+            liStyle = xbmcgui.ListItem(startTime + " " + title,
+                thumbnailImage=thumb)
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Drop

File: plugin.video.raitv/default.py [hunk 12]
```diff
@@ -0,0 +1,525 @@
...
+        
+        if programme["hasAudio"]:
+            audioUrl = programme["pathID"]
+        else:
+            audioUrl = None
+        
+        if audioUrl is None:
+            # programme is not available
+            liStyle = xbmcgui.ListItem(startTime + " [I]" + title + "[/I]",
+                thumbnailImage=thumb)
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Drop

File: plugin.video.raitv/default.py [hunk 13]
```diff
@@ -0,0 +1,525 @@
...
+        show_tgr_root()        
+elif mode == "get_last_content_by_tag":
+     get_last_content_by_tag(tags)
+elif mode == "get_most_visited":
+     get_most_visited(tags)
+
+elif mode == "play":
+    play(url, pathId)
+
+else:
```
Comment (enen92) [inline plugin.video.raitv/default.py]: Usually a good practice to keep the entrypoint as small as possible

File: plugin.video.raitv/default.py (hunk 10 above)
Comment (maxbambi) [reply plugin.video.raitv/default.py]: These strings are used to search in json, so they aren't string in GUI

Comment (enen92) [reply plugin.video.raitv/default.py]: Disregard, marking as solved.

**Comment by maxbambi** [comment]
i'm waiting for new checks of code... previous version of plugin has disappeared in repo, so i don't know where and how to mark it as broken

**Comment by basrieter** [comment]
Please make this PR a single commit with the proper naming for that commit.

**Comment by maxbambi** [comment]
> Please make this PR a single commit with the proper naming for that commit.

**Comment by maxbambi** [comment]
@basrieter, There is a new PR with a single commit

**Pull request closed without merging**